src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx

```diff
@@ -58,166 +58,178 @@ interface YearSelection {
   };
 }
 
+// Initialize data structure for displaying units by year
+let yearData: {
+  [key: string]: {
+    units: Unit[];
+    childSubjects: Subject[];
+    domains: Domain[];
+    tiers: Tier[];
+  };
+} = {};
+let threadOptions: Thread[] = [];
+let yearOptions: string[] = [];
+const unitSlugs = new Set<string>();
+const duplicateUnitSlugs = new Set<string>();
+
 // Function component
 
 const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
   // Initialize constants
 
-  const threadOptions: Thread[] = [];
-  const yearOptions: string[] = [];
   const { track } = useAnalytics();
   const { analyticsUseCase } = useAnalyticsPageProps();
   const [displayModal, setDisplayModal] = useState(false);
   const [unitData, setUnitData] = useState<Unit | null>(null);
   const [unitOptionsAvailable, setUnitOptionsAvailable] =
     useState<boolean>(false);
   const modalButtonRef = useRef<HTMLButtonElement>(null);
-  const unitSlugs = new Set<string>();
-  const duplicateUnitSlugs = new Set<string>();
 
-  // Initialize data structure for displaying units by year
+  const [yearSelection, setYearSelection] = useState<YearSelection>({});
+  const [selectedThread, setSelectedThread] = useState<Thread | null>(null);
+  const [selectedYear, setSelectedYear] = useState<string | null>(null);
 
-  const yearData: {
-    [key: string]: {
-      units: Unit[];
-      childSubjects: Subject[];
-      domains: Domain[];
-      tiers: Tier[];
-    };
-  } = {};
+  // Put data formatting code in useEffect to avoid unnecessary re-renders
+  useEffect(() => {
```

Comment:
So Carl has made me newly averse to useEffect. It's basically not recommended for use unless connecting to an external system . See here:

https://react.dev/reference/react/useEffect#connecting-to-an-external-system

Without a deeper look I'm not sure what would be needed to avoid it in this scenario, but it shouldn't be necessary.